### PR TITLE
Fix Shell Title Calculation

### LINKED ISSUE #3382: [Bug] [Android] Title of Shell initially not set to correct value
### Description
The initial value of the `Shell`s title is not set to the title of the `ShellContent`. Instead it displays the name of the application. When navigating to another page, the toolbar displays the title of the new page.

### Steps to Reproduce

1. Create a `Shell`
2. Add two `FlyoutItem`s with some pages where the title is set
3. Start the application (the text in the toolbar does not match the pages title)
4. Navigate to another page
5. The toolbar displays the correct title

### Expected Behavior
The title displayed in the toolbar does match the title of the initial page.

### Actual Behavior
The title displayed in the toolbar does not match the title of the initial page. Instead the name of the application gets displayed

### Basic Information

- Version with issue: 6.0.101-preview.10.2068
- Last known good version: -
- IDE: Visual Studio 2022 Version 17.1.0 Preview 1.0
- Platform Target Frameworks: 
  - Android: 28

#### Expected
image

#### Actual
image

### Reproduction Link
[Here](https://github.com/LarsBehl/StatusApp/blob/9554fb14bc9aa497eb4697c2f790b4bb87b45c9b/SideNav.xaml) is the link to a `Shell` where the issue is present. The used pages can also be found in this repository.

### LINKED ISSUE #4310: The App using Shell always have App Name as default title.
### Description

The App using Shell always have App Name as default title. Although I set the title on AppShell as well as on MainPage in the reproduction but still the Title always remain the AppName. 

### Steps to Reproduce

Navigate to a page through Shell. The title of the navigated page will be the name of the App. 
[BugSample.zip](https://github.com/dotnet/maui/files/8031158/BugSample.zip)

### Version with bug

Preview 12 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 11.0- API 30

### Did you find any workaround?

Even TitleView does not resolve the issue. Hide NavigationBar with Shell.SetNavBarIsVisible(this, false); and Create your own view. 

### Relevant log output

_No response_

### LINKED ISSUE #5105: [Android] Shell AppBar Title is set and read aloud twice
### Description

On Android, when using Shell, the AppBar title is set (it is unclear where the title is being set from) and is also read aloud twice (rather than just once) by TalkBack

### Steps to Reproduce

1. Open [TestPR4524.zip](https://github.com/dotnet/maui/files/8200843/TestPR4524.zip)
2. Run on an Android device - notice "TestPR4524" title is set in app bar
3. Turn on TalkBack - notice the title "TestPR4524" is announced twice

Screenshot_20220307-150834

### Version with bug

Preview 13 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 11

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change

- Fix Shell to walk up it's hierarchy to locate a title to use
- We switched to using ViewPager2 with Shell and incorrectly used the `ConfigureTab` method on the mediator to process selected index changes. This PR adds back the PageChanged listener which correctly sets selected item now

### Issues Fixed

Fixes #5105

### CODE REVIEW

File: src/Controls/src/Core/Shell/Shell.cs [hunk 1]
```diff
@@ -1224,19 +1224,28 @@ internal FlyoutBehavior GetEffectiveFlyoutBehavior()
...
-			element = element ?? GetVisiblePage() ?? CurrentContent;
+			element = element ?? GetCurrentShellPage() ?? CurrentContent;
 			while (element != this && element != null)
 			{
 				observer?.Invoke(element);
 
-				if (element.IsSet(property))
+				if (ignoreImplicit && Routing.IsImplicit(element))
+				{
+					// Ignore Implicit routes
```
Comment (rachelkang) [inline src/Controls/src/Core/Shell/Shell.cs:1246]: Elaborate more on comment for future understanding

**Review by rachelkang — APPROVED** [review]
LGTM now once it's all green! 
Thanks for hearing my feedback and making all those tweaks these past few days

**Comment by jfversluis** [comment]
As per the [release notes](https://github.com/dotnet/maui/releases/tag/6.0.200-preview.14.2) this fix was not included in that release so it will be part of the next one :)

**Pull request merged**